### ISSUE TITLE
Hive DataFrame columns and push errors

### ISSUE THREAD

**Issue by naoyak** [airbnb/omniduct#68]
`HiveServer2Client` has been acting weirdly since `0.9.0` and I can't seem to track down what the root causes are. Not sure if this is a config error. Any ideas @matthewwardrop ?

1. Table name appended to columns when querying `SELECT * FROM...`
When querying with `SELECT * FROM <table>`, the returned dataframe has table names appended with a `.`
![screen shot 2018-09-13 at 3 45 22 pm](https://user-images.githubusercontent.com/3279641/45519905-0d29e300-b76c-11e8-99d9-b6b6d673a8a8.png)

2. `HiveServer2Client.dataframe_to_table()` not working
```
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
<ipython-input-8-8116e3eff828> in <module>()
----> 1 ap.hive.dataframe_to_table(df1, table='naoya.some_temp_table')

<decorator-gen-155> in dataframe_to_table(self, df, table, if_exists, **kwargs)

~/miniconda/envs/airpydev/lib/python3.6/site-packages/omniduct/duct.py in wrapped(f, *args, **kw)
     68         @decorator.decorator
     69         def wrapped(f, *args, **kw):
---> 70             return f(*args, **kw)
     71 
     72         mro = inspect.getmro(cls)

<decorator-gen-147> in dataframe_to_table(self, df, table, if_exists, **kwargs)

~/miniconda/envs/airpydev/lib/python3.6/site-packages/omniduct/utils/debug.py in logging_scope(func, *args, **kwargs)
    270         except Exception as e:
    271             success = False
--> 272             raise_with_traceback(e)
    273         finally:
    274             logger._scope_exit(success)

~/.local/lib/python3.6/site-packages/future/utils/__init__.py in raise_with_traceback(exc, traceback)
    417         if traceback == Ellipsis:
    418             _, _, traceback = sys.exc_info()
--> 419         raise exc.with_traceback(traceback)
    420 
    421 else:

~/miniconda/envs/airpydev/lib/python3.6/site-packages/omniduct/utils/debug.py in logging_scope(func, *args, **kwargs)
    266         success = True
    267         try:
--> 268             f = func(*args, **kwargs)
    269             return f
    270         except Exception as e:

~/miniconda/envs/airpydev/lib/python3.6/site-packages/omniduct/databases/base.py in dataframe_to_table(self, df, table, if_exists, **kwargs)
    599         """
    600         assert if_exists in {'fail', 'replace', 'append'}
--> 601         self.connect()._dataframe_to_table(df, self._parse_namespaces(table), if_exists=if_exists, **kwargs)
    602 
    603     # Table properties

~/miniconda/envs/airpydev/lib/python3.6/site-packages/omniduct/databases/hiveserver2.py in _dataframe_to_table(self, df, table, if_exists, schema, use_hive_cli, partition, sep, table_props, dtype_overrides, **kwargs)
    356             proc = self._run_in_hivecli(stmts)
    357             if proc.returncode != 0:
--> 358                 raise RuntimeError(proc.stderr.decode('utf-8'))
    359         finally:
    360             # Clean up files

RuntimeError: /bin/sh: naoya: command not found
/bin/sh: some_temp_table: command not found
/bin/sh: naoya: command not found
/bin/sh: some_temp_table: command not found
18/09/13 22:46:25 WARN conf.HiveConf: DEPRECATED: Configuration property hive.metastore.local no longer has any effect. Make sure to provide a valid value for hive.metastore.uris if you are connecting to a remote metastore.

Logging initialized using configuration in jar:file:/mnt/var/opt/CDH-5.3.3-1.cdh5.3.3.p2826.3087/jars/hive-common-0.13.1-cdh5.3.3.jar!/hive-log4j.properties
NoViableAltException(17@[184:1: tableName : (db= identifier DOT tab= identifier -> ^( TOK_TABNAME $db $tab) |tab= identifier -> ^( TOK_TABNAME $tab) );])
	at org.antlr.runtime.DFA.noViableAlt(DFA.java:158)
	at org.antlr.runtime.DFA.predict(DFA.java:144)
	at org.apache.hadoop.hive.ql.parse.HiveParser_FromClauseParser.tableName(HiveParser_FromClauseParser.java:4927)
	at org.apache.hadoop.hive.ql.parse.HiveParser.tableName(HiveParser.java:40229)
	at org.apache.hadoop.hive.ql.parse.HiveParser.createTableStatement(HiveParser.java:4408)
	at org.apache.hadoop.hive.ql.parse.HiveParser.ddlStatement(HiveParser.java:2138)
	at org.apache.hadoop.hive.ql.parse.HiveParser.execStatement(HiveParser.java:1392)
	at org.apache.hadoop.hive.ql.parse.HiveParser.statement(HiveParser.java:1030)
	at org.apache.hadoop.hive.ql.parse.ParseDriver.parse(ParseDriver.java:199)
	at org.apache.hadoop.hive.ql.parse.ParseDriver.parse(ParseDriver.java:166)
	at org.apache.hadoop.hive.ql.Driver.compile(Driver.java:417)
	at org.apache.hadoop.hive.ql.Driver.compile(Driver.java:335)
	at org.apache.hadoop.hive.ql.Driver.compileInternal(Driver.java:1026)
	at org.apache.hadoop.hive.ql.Driver.runInternal(Driver.java:1091)
	at org.apache.hadoop.hive.ql.Driver.run(Driver.java:962)
	at org.apache.hadoop.hive.ql.Driver.run(Driver.java:952)
	at org.apache.hadoop.hive.cli.CliDriver.processLocalCmd(CliDriver.java:269)
	at org.apache.hadoop.hive.cli.CliDriver.processCmd(CliDriver.java:221)
	at org.apache.hadoop.hive.cli.CliDriver.processLine(CliDriver.java:431)
	at org.apache.hadoop.hive.cli.CliDriver.processLine(CliDriver.java:367)
	at org.apache.hadoop.hive.cli.CliDriver.executeDriver(CliDriver.java:750)
	at org.apache.hadoop.hive.cli.CliDriver.run(CliDriver.java:694)
	at org.apache.hadoop.hive.cli.CliDriver.main(CliDriver.java:633)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.apache.hadoop.util.RunJar.main(RunJar.java:212)
FAILED: ParseException line 2:47 cannot recognize input near 'naoya_kanai' '.' '.' in table name
```

**Comment by naoyak** [comment]
My sense is that the query is somehow malformed. Can anyone replicate on their Hive setup?

**Comment by matthewwardrop** [comment]
@naoyak Sorry! Didn't see this earlier.

1: This is just the behaviour of hive (as compared to presto). Field names are prefixed with the table/subquery from which they are drawn unless manually renamed. This is not a bug.

2: Looks like there is some issue with the `hive` command on the remote host. Can you increase the decrease the logging level so we can see what's actually being sent? Just run `omniduct.utils.debug.logger.logging_level = 1`, and then rerun the above.

**Comment by naoyak** [comment]
@matthewwardrop I opened a fix at #76. Not sure this is the best solution but works (when accessing remotely).

**Comment by matthewwardrop** [comment]
Closing in favour of discussion at #76
